src/components/tabs/index.tsx

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useState } from 'react'
+
+interface TabsProps {
+  active: number
+  tabs: Array<{
+    children: React.ReactNode
+    name: string
+  }>
+}
+
+function Tabs({ tabs, active }: TabsProps) {
+  const [activeTab, setActiveTab] = useState(active)
+
+  useEffect(() => {
+    setActiveTab(active)
+  }, [active])
```

Comment:
Manter 2 variaveis para orientar o mesmo estado não me parece uma estrategia inteligente, pois vai gerar uma não sincronia com o estado externo, se queremos que o componente pai tenha controle do estado, então somente ele deve ter esse state e enviar o state e onChange para o filho. O que acha?
```suggestion
import { useState } from 'react'

interface TabsProps {
  initialTab: number
  tabs: Array<{
    children: React.ReactNode
    name: string
  }>
}

function Tabs({ tabs, initialTab }: TabsProps) {
  const [activeTab, setActiveTab] = useState(initialTab)
```